tools/bin/ariaDownload.py

```diff
@@ -260,43 +257,55 @@ def make_nc_cookie(self):
        log.info('\n\nNew users: you must first log into Vertex and accept the EULA. In addition, your Study Area must be set at Earthdata https://urs.earthdata.nasa.gov')
        os.sys.exit(1)
 
-def prod_dl(nt):
+def prod_dl(inps, dct_prod):
     """ Perform downloading using ASF bulk dl; parallel processing supported """
     import multiprocessing
-    import ASFDataDload as AD
-    args = os.sys.argv
-    os.sys.argv = [] # gets around spurious messages
-    downloader  = AD.bulk_downloader()
+    # import ASFDataDload as AD
+    # args = os.sys.argv
+    # os.sys.argv = [] # gets around spurious messages
+    # downloader  = AD.bulk_downloader()
+
+
     max_threads = multiprocessing.cpu_count()
-    os.sys.argv = args # required for pool
+    # os.sys.argv = args # required for pool
 
-    if nt == 'all':
+    if inps.num_threads == 'all':
         nt = max_threads
     else:
-        nt = int(nt)
+        nt = int(inps.num_threads)
 
     nt = nt if nt < max_threads else max_threads
     log.info('Using  %s threads for parallel downloads', nt)
-    chunks = np.array_split(downloader.files, nt)
-    chunks = [chunk for chunk in chunks if chunk.size > 0]
-    chunk_check = []
-    # sanity check
-    for chunk in chunks: chunk_check.extend(chunk)
-    if not chunk_check == downloader.files:
-        log.error('Numpy incorrectly split product list; run serially')
-        raise RuntimeError
-
-    # the ASF downloader obj must be called again in here otherwise multiprocessing fails
-    with multiprocessing.Pool(nt) as pool:
-         pool.map(_dl_helper, chunks)
-
-def _dl_helper(files):
+    files   = dct_prod['product_list'].split(',')
+    chunks1 = np.array_split(files, np.ceil(len(files)/200)) # split by 200s
+    check   = []
+    for chunk in chunks1:
+        chunks     = np.array_split(chunk, nt)
+        lst_dcts   = [vars(inps)]*nt  # Namespace->dctionary, repeat it
+        for i in range(len(chunks)): # put split up files to the objects for threads
```

Comment:
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)